sdk/servicebus/azure-servicebus/azure/servicebus/amqp/_amqp_message.py

```diff
@@ -0,0 +1,450 @@
+            raise ValueError(
+                "There should be one and only one of either data_body, sequence_body "
+                "or value_body being set as the body of the AMQPAnnotatedMessage."
+            )
+
+        header = kwargs.get("header")
+        footer = kwargs.get("footer")
+        properties = kwargs.get("properties")
+        application_properties = kwargs.get("application_properties")
+        annotations = kwargs.get("annotations")
+        delivery_annotations = kwargs.get("delivery_annotations")
+
+        body_type = (
+            uamqp.MessageBodyType.Data
+            if self._data_body
+            else (
+                uamqp.MessageBodyType.Sequence
+                if self._sequence_body
+                else uamqp.MessageBodyType.Value
+            )
+        )
+        body = self._data_body or self._sequence_body or self._value_body
+
+        message_header = None
+        message_properties = None
+        if header:
+            message_header = uamqp.message.MessageHeader()
+            message_header.delivery_count = header.get("delivery_count", 0)
+            message_header.time_to_live = header.get("time_to_live")
+            message_header.first_acquirer = header.get("first_acquirer")
+            message_header.durable = header.get("durable")
+            message_header.priority = header.get("priority")
+
+        if properties:
+            message_properties = uamqp.message.MessageProperties(**properties)
+
+        self._message = uamqp.message.Message(
+            body=body,
+            body_type=body_type,
+            header=message_header,
+            footer=footer,
+            properties=message_properties,
+            application_properties=application_properties,
+            annotations=annotations,
+            delivery_annotations=delivery_annotations,
+        )
+
+    @property
+    def body(self):
+        # type: () -> Any
+        """The body of the Message. The format may vary depending
+        on the body type:
+        For ~azure.servicebus.AMQPMessageBodyType.DATA, the body could be bytes or Iterable[bytes]
+        For ~azure.servicebus.AMQPMessageBodyType.SEQUENCE, the body could be List or Iterable[List]
+        For ~azure.servicebus.AMQPMessageBodyType.VALUE, the body could be any type.
+
+        :rtype: Any
+        """
+        return self._message.get_data()
+
+    @property
+    def body_type(self):
+        # type: () -> Optional[AMQPMessageBodyType]
+        """The body type of the underlying AMQP message.
+
+        rtype: Optional[~azure.servicebus.amqp.AMQPMessageBodyType]
+        """
+        return AMQP_MESSAGE_BODY_TYPE_MAP.get(
+            self._message._body.type  # pylint: disable=protected-access
+        )
+
+    @property
+    def properties(self):
+        # type: () -> Optional[AMQPMessageProperties]
+        """
+        Properties to add to the message.
+
+        :rtype: Optional[~azure.servicebus.amqp.AMQPMessageProperties]
```

Comment:
in GAed it's returning `uamqp.message.MessageProperties`. This would be a breaking change to stuff like type hint / isinstance check. but we don't know how much existing users reply on the type check (as far as I understand, raw_amqp_message should be a rare use case).

way to hack isinstance:
```python
    @property
    def __class__(self):
        return uamqp.message.MessagePropreties  # bad idea?
```
or
```
class AMQPMessageProperties(DicMixin, uamqp.message.MessageProperties)
```

Reply:
is the first way (with setting __class__ property to return `uamqp.message.MessageProperties`) the one that would be a breaking change?

Reply:
@yunhaoling @annatisch I'd like to get rid of everything that shows `uamqp`, in case we replace that transport by something else. As long as the duck typing is consistent, I think I'm happy enough.